### ISSUE TITLE
Purchase, Purchase Items and Product Variants Feed not working

### ISSUE THREAD

**Issue by pip77** [exponea/bloomreach-magento2-integration#44]
Hi,

We have activated the plugin and were able to do the initial import of the Products and Customer Feeds, but unfortunately the Initial Imports of Purchase, Purchase Items and Product Variants Feed are not working. After starting the feeds, they are stuck at PROCESSING and not showing anything inside the Progress Log. Also I was not able to find any errors in the cron.log, exception.log or bloomreach/debug.log.

Is there a way to have a more detailed log or do you have an idea where I can look for errors? Are the three stuck feeds maybe connected? Maybe the product_id? As far as I can see the product_id is not used in the Customer and Products Feed.

We are using Magento ver. 2.4.3.

Appreciate your help!

![image](https://github.com/exponea/bloomreach-magento2-integration/assets/11523057/e0651b3b-30d8-448a-9c36-fa05f7a08769)
![image](https://github.com/exponea/bloomreach-magento2-integration/assets/11523057/afb439df-e219-45be-8c16-5c2aa170ad48)

**Comment by pip77** [comment]
It seems like the jobs stop after 5 min. Is there any execution limit we need to configure?

**Comment by vlmed** [comment]
@pip77 
it seems to be memory issue.

Check server logs (nginx or apache) for relevant errors. Like: Fatal error: Allowed memory size of 134217728 bytes exhausted. 

The fix has been already implemented and will be available in the next release.
As a temporary solution you can manually apply a fix:
https://patch-diff.githubusercontent.com/raw/exponea/bloomreach-magento2-integration/pull/42.patch

**Comment by pip77** [comment]
@vlmed also, if I set the PAGE_SIZE to 1 for exmaple?

**Comment by pip77** [comment]
@vlmed what makes me curious is that it is working for 120k Customers, but not for 50k Products.

**Comment by vlmed** [comment]
@pip77 
Actually this is an interesting question. 
It might be related to some customizations or third party modules or whatever.
We tested the extension with more than 1 million products and the import went well.
Please try to apply the fix and run the import again. Most likely, this will solve the problem.

**Comment by pip77** [comment]
Thank you both. We will try.

**Comment by pip77** [comment]
@vlmed @eduard13 thank you for your help. We have found the problem. The crons where killed after 5 minutes by the server. We have adjusted this configuration. 

After increasing the allowed memory, the max runtime of the crons and fixing and fixing the error 
> Return value of Bloomreach\EngagementConnector\Model\DataMapping\FieldValueRenderer\Order\ProductList::getProductSku() must be of the type string, null returned

I have created a new issue for the error above: https://github.com/exponea/bloomreach-magento2-integration/issues/46

Everything seems to work as expected. We can close this issue here I guess.
